Shopipy.AppointmentManagement/Services/AppointmentService.cs

```diff
@@ -0,0 +1,187 @@
+using Shopipy.Persistence.Models;
+using Shopipy.Persistence.Repositories;
+using System.Linq;
+using System.Runtime.InteropServices.JavaScript;
+
+namespace AppointmentManagement.Services;
+
+public class AppointmentService
+{
+    private readonly IGenericRepository<Appointment> _appointmentRepository;
+    private readonly IGenericRepository<Service> _serviceRepository;
+    private readonly IGenericRepository<User> _userRepository;
+
+    public AppointmentService(IGenericRepository<Appointment> appointmentRepository,
+        IGenericRepository<Service> serviceRepository, IGenericRepository<User> userRepository)
+    {
+        _appointmentRepository = appointmentRepository;
+        _serviceRepository = serviceRepository;
+        _userRepository = userRepository;
```

Comment:
Do not access userRepository directly, use userService (PR not merged yet :( )

Reply:
userService only has one method ( getCurrentUserAsync ), I'll wait for the PR to be merged : |